week-4/Homework/mandatory/4-groceries.js

```diff
@@ -30,6 +30,22 @@ Exercise 1:
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
 
+// weeklyMealPlan.forEach(function (day) {
+//   if (day.length > 0) {
+//     for (let i = 0; i <day.length; i++) {
+//       weeklyGroceriesToBuy.push(day[i])
+//     }
+//   }
+// })
+
+for (day in weeklyMealPlan) {
+  if (weeklyMealPlan[day].length > 0) {   
```

Comment:
Don't need to check if array is empty twice. You have the same check implemented in the condition inside the for-loop.

Reply:
I didn’t understand what you mean here.

Reply:
So technically these are the same if you think about that
`if (weeklyMealPlan[day].length > 0)` === `for (let i = 0; i < weeklyMealPlan[day].length; i++)`

If the array is empty then the for-loop will do nothing, it is redundant to check in the if statement too.

Reply:
fixed it. I think I'm tip-toing when writing code, I hate to get an error message.